lib/client/product.rb

```diff
@@ -55,11 +55,15 @@ def upsert(payload, params={})
 
             # When using sku:in you must specify the fields you want returned.
             def get_by_skus(skus, include = %w[custom_fields modifiers], include_fields = %w[sku])
-                products = index({
-                    'sku:in': skus.join(','),
-                    include: include.join(','),
-                    include_fields: include_fields.join(','),
-                })
+                products = []
+                skus.each do |sku|
+                    data = index({
+                        'sku': sku,
+                        include: include.join(','),
+                        include_fields: include_fields.join(','),
+                    })
+                    products.push(data[0])
+                end
```

Comment:
@Jhalter5Stones Because we're pulling each product individually, we may want to rework how we're handling errors from this function.

We probably want the error event to highlight which SKU caused the issue, so the initial _catch_ should happen in this method instead of the agent method that _calls_ this.

(Consequently, you should rebase off of master to get the updated error handling. The error thrown here should be:

```
BigCommerceProductError.new(
    status,
    'get_by_skus',
    'Failed to get existing product data',
    sku,
    data: { related_skus: [ ... ] }
)
```
Then modify the product agent accordingly to catch this error, emit an error event, and rethrow the exception as it does currently.

Reply:
@mrogers-5s So like this?